tests/Helper/ResponseHelper.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2013-2017 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Test\Helper;
+
+use Symfony\Component\HttpFoundation;
+
+/**
+ * @mixin \PHPUnit\Framework\TestCase
+ */
+trait ResponseHelper
```

Comment:
This trait is currently only used by the `WebTestCase`, but I am hesitant to pull it up into the `WebTestCase` itself, as I feel the assertions should not be able to access any internal state of `WebTestCase` and children of it.

Reply:
This turned out similar to what I expected. There is a lot of repetition in readability of tests because we've extracted, more or less a stateless procedural interface for assertions on `Response`. I think that any `WebTestCase` (or Behat `FeatureContext`, more aptly named) carries with it a usefully stateful context, by definition. 

It's the difference between:

```php
# Current behaviour and arguably ambiguous. Is this an assertion? How do I know?
$this->get('/')->isSuccessful();
```
and...
```php
// Okay, makes sense.
$response = $this->get('/');

// I feel like I'm typing response over and over again. 
// It's mechanical-feeling but is 100% an assertion.
$this->assertResponseIsSuccessful($response); // response is response is response is.
```

Personally (and I think this is really just preferences honestly) I'm fine with decorating a response in tests and making assertions through it. It's 6 vs a half-dozen compared to a procedural interface through trait, in my opinion.

It's certainly readable. What if we removed the ambiguity of "is this an assertion"? Something like:

```php
// Normal get request.
$this->get('/')
    ->assertSuccessful()
    ->assertSee('OpenCon 2017');

// I'm creating a resource via API (D:)
$this->post('/users', [ /* data */ ])
    ->assertCreated();   // vs assertSuccessful(Response::HTTP_CREATED)
```

If we're definitely moving away from a decorator with assertions, I am currently leaning more towards stateful assertions on `WebTestCase`, itself to maximize readability. Could reset `$this->response` on `call(...)` and everything would be good to go. 

To completely flip the table, I really do feel like the original purpose of `WebTestCase` was to provide a more-human interface for making high-level behavioural assertions from and end-to-end perspective. Like I've said previously, it may just be time to actually make that move.